### DISCUSSION TITLE
How to put images in a table?

### DISCUSSION THREAD

**Discussion by ceprio** [nocodb/nocodb#709] [Q&A] [answered]
I'm trying to figure out how to have an image inserted in a cell of a table or a column of type "image". I'm not finding any help on this on the web. Where to start?
My end goal is to have a form where an image can be uploaded along with a description of the image.

Any clue on how to acheive this whould be appreciated.

Great project by the way, I've been looking for something like this for a long time!

**Comment by wingkwong** [comment, marked as answer]
You can create a column with ``Attachment`` type.

![image](https://user-images.githubusercontent.com/35857179/139478493-b8c28ae5-46e3-48a7-8c60-0c75c4b54205.png)

If you wanna store the images in Amazon S3, you can go to App Store 

![image](https://user-images.githubusercontent.com/35857179/139478581-7df26e0f-1e9a-45f5-9dfe-c04733a2091e.png)

and install S3 plugin

![image](https://user-images.githubusercontent.com/35857179/139478680-8da7307d-c0de-4d06-b522-63e7071757b3.png)

**Comment by yologith** [comment]
I'm facing the same problem to insert an image in to "image" type column by REST API.

(I know it's not a good idea to store image blob in db and the "image" type had deprecated, it's legacy project I've to handle though.)

I've tried to use GET API method for retrieving an image that already been in the table and I've got JSON object as a result.

`{
        "type": "Buffer",
        "data": [
            137,
            80,
            78,
            71,
            13,
            10,
            ..........
        ]
    }`

I've also tried to use the same format in POST API method to insert a new row and got error message.

`Operand type clash: nvarchar is incompatible with image`

Any advice would be appreciated.
